crowbar_framework/app/models/updater_service.rb

```diff
@@ -75,6 +124,20 @@ def apply_role_pre_chef_call(old_role, role, all_nodes)
     @logger.debug("Updater apply_role_post_chef_call: exiting")
   end
 
+  def apply_role_post_chef_call(old_role, role, all_nodes)
+    all_nodes.each do |n|
+      node = NodeObject.find_node_by_name n
+      if node.role? "updater-upgrade"
+        role_name = NodeObject.make_role_name(node.name)
+        node.run_list.run_list_items.clear
+        node.run_list.run_list_items << "role[#{role_name}]"
+        node.crowbar["state"] = "upgrade"
```

Comment:
Thinking out loud: I'm thinking we want to split that upgrade state in two: crowbar_upgrade and os_upgrade. What we're doing in this PR would be crowbar_upgrade, and when crowbar is then upgraded and the user says "okay, I want to now upgrade all nodes too", we move to "os_upgrade" which triggers autoyast profile generation, reboot of nodes, etc.